pkg/image/registry/image/strategy.go

```diff
@@ -129,6 +129,38 @@ func (imageStrategy) clearSignatureDetails(image *api.Image) {
 	}
 }
 
+// internalStrategy implements behaviour for signatures subset of image.
+type internalStrategy struct {
+	imageStrategy
+}
+
+// InternalStrategy is the default logic that applies when updating
+// image' signatures via the REST API.
+var InternalStrategy = internalStrategy{Strategy}
+
+// Canonicalize normalizes the object after validation.
+func (internalStrategy) Canonicalize(obj runtime.Object) {
+}
+
+// PrepareForUpdate currently allows to set just image signatures.
+func (s internalStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	oldImage := old.(*api.Image)
+	image := obj.(*api.Image)
+
+	image.Labels = oldImage.Labels
+
+	image.DockerImageReference = oldImage.DockerImageReference
```

Comment:
This is unsafe because you could unintentionally forget copying something new.  Do it where only things explicitly known are preserved in new.